packages/plugins/graphql/server/src/bootstrap.ts

```diff
@@ -20,6 +20,86 @@ const merge = mergeWith((a, b) => {
   }
 });
 
+export const determineLandingPage = (strapi: Core.Strapi) => {
+  const { config } = strapi.plugin('graphql');
+  const utils = strapi.plugin('graphql').service('utils');
+
+  /**
+   * configLanding page may be one of the following:
+   *
+   * - true: always use "playground" even in production
+   * - false: never show "playground" even in non-production
+   * - undefined: default Apollo behavior (hide playground on production)
+   * - a function that returns an Apollo plugin that implements renderLandingPage
+   ** */
+  const configLandingPage = config('landingPage');
+
+  const isProduction = process.env.NODE_ENV === 'production';
+
+  const localLanding = () => {
+    strapi.log.debug('Apollo landing page: local');
+    utils.playground.setEnabled(true);
+    return ApolloServerPluginLandingPageLocalDefault();
+  };
+
+  const prodLanding = () => {
+    strapi.log.debug('Apollo landing page: production');
+    utils.playground.setEnabled(false);
+    return ApolloServerPluginLandingPageProductionDefault();
+  };
+
+  const userLanding = (userFunction: (strapi?: Core.Strapi) => ApolloServerPlugin | boolean) => {
+    strapi.log.debug('Apollo landing page: from user-defined function...');
+    const result = userFunction(strapi);
+    if (result === true) {
+      return localLanding();
+    }
+    if (result === false) {
+      return prodLanding();
+    }
+    strapi.log.debug('Apollo landing page: user-defined');
+    return result;
+  };
+
+  // DEPRECATED, remove in Strapi v6
```

Comment:
Should we update the documentation somewhere to add a callout about this?

Reply:
Or is the warning enough?

Reply:
I have a docs PR too: https://github.com/strapi/documentation/pull/2301